controllers/helpers.go

```diff
@@ -32,6 +32,78 @@ import (
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
+// filterContainerIDs filters out any containers that may not be intended for disruption in a multi-container disruption
+// this function returns 3 string slices:
+// containers: The updated list of containers with the filtered out containers removed
+// removed: The list of containers that were removed from the list containers, used later for logging purposes
+// reasoning: A list of strings used later for logging the reason as to why the corresponding container in removed was removed
+func filterContainerIDs(pod *corev1.Pod, containers []string, spec v1beta1.DisruptionSpec) ([]string, []string, []string) {
+	removed := []string{}
+	reasoning := []string{}
+
+	if spec.DiskPressure != nil {
+		// validate that each container has the volume to be disrupted
+		removed, reasoning = getNoValidVolumeCtns(pod, spec)
+		containers = removeCtnsFromConsideration(pod, removed, containers)
+	}
+
+	return containers, removed, reasoning
+}
+
+// removeCtnsFromConsideration removes containers from the removal list that do not comply with multi-container
+// disruption rules. Returns the updated list of containers
+func removeCtnsFromConsideration(pod *corev1.Pod, removal []string, containers []string) []string {
+	if len(removal) == 0 {
+		return containers
+	}
+
+	for _, ctn := range pod.Status.ContainerStatuses {
```

Comment:
I also cannot find a way to avoid this triple embedded for loop, given that volume mounts are in pod.Spec but containerID is in pod.Status

Reply:
I am not a fan of it either :/